sway/desktop/output.c

```diff
@@ -651,4 +673,72 @@ void handle_new_output(struct wl_listener *listener, void *data) {
 	}
 
 	transaction_commit_dirty();
+
+	update_output_manager_config(server);
+}
+
+void handle_output_manager_apply(struct wl_listener *listener, void *data) {
+	struct sway_server *server =
+		wl_container_of(listener, server, output_manager_apply);
+	struct wlr_output_configuration_v1 *config = data;
+
+	struct wlr_output_configuration_head_v1 *config_head;
+	// First disable outputs we need to disable
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+		struct sway_output *output = wlr_output->data;
+		if (!output->enabled || config_head->state.enabled) {
+			continue;
+		}
+		struct output_config *oc = new_output_config("temp");
+		oc->enabled = false;
+		// There is no explicit way to set DPMS through the protocol.
+		// Users probably expect that disabling a monitor will turn it off.
+		oc->dpms_state = DPMS_OFF;
+
+		apply_output_config(oc, output);
+		free_output_config(oc);
+	}
+
+	// Then enable outputs that need to
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+		struct sway_output *output = wlr_output->data;
+		if (!config_head->state.enabled) {
+			continue;
+		}
+		struct output_config *oc = new_output_config("temp");
```

Comment:
Instead of "temp" we could use the output name? I don't remember what this string is used for TBH.

Reply:
Doesn't seem to be relevant here as far as I can tell - I used temp as that was what I saw elsewhere in the code.

Reply:
It's used when merging output configs in (`store_output_config`) and determining which output to apply the config to (in `apply_output_config_to_outputs`). Currently, this config is not being stored so if the output command is ran for the output (via swaymsg or a binding) it will revert everything set by this config to whatever is stored

Reply:
I guess we need to store the output config too then. Not sure which identifier to use though.